File: backend/app/schemas/replicacao.py
from pydantic import Field
from app.schemas.common import ORMBase


class ReplicarLancamentos(ORMBase):
    ano_origem_id: int
    mes_origem: int = Field(ge=1, le=12)
    ano_destino_id: int
    mes_destino: int = Field(ge=1, le=12)
    replicar_receitas: bool = True
    replicar_despesas: bool = True
    apenas_recorrentes: bool = True
    force: bool = False  # sobrescrever se já houver no destino


class ResultadoReplicacao(ORMBase):
    receitas_replicadas: int
    despesas_replicadas: int
    receitas_origem_total: int
    despesas_origem_total: int
    mensagem: str

File: frontend/src/services/replicacao.ts
import { http } from "./http"

export interface ReplicarPayload {
  ano_origem_id: number
  mes_origem: number
  ano_destino_id: number
  mes_destino: number
  replicar_receitas: boolean
  replicar_despesas: boolean
  apenas_recorrentes: boolean
  force: boolean
}

export interface ResultadoReplicacao {
  receitas_replicadas: number
  despesas_replicadas: number
  receitas_origem_total: number
  despesas_origem_total: number
  mensagem: string
}

export const replicacaoService = {
  replicar: (p: ReplicarPayload) =>
    http.post<ResultadoReplicacao>("/lancamentos/replicar", p).then(r => r.data),
}